RpcParser/markdown_generator.py

```diff
@@ -0,0 +1,225 @@
+                    if isinstance(item, Function):
+                        mark_down.write('\nMessage Type: **{}**\n'.format(item.message_type.name))
+                    if item.description:
+                        mark_down.write('\n{}\n'.format(re.sub(r'\s{2,}', ' ', ''.join(item.description)).strip()))
+
+                    if getattr(item, container_name):
+                        mark_down.write('\n##### {}\n\n| Value | '.format(elem_or_param))
+                        if isinstance(item, Enum):
+                            mark_down.write('Description | \n| ---------- |:-----------:|')
+                        else:
+                            mark_down.write(' Type | Mandatory | Description | \n'
+                                            '| ---------- | ---------- |:-----------: |:-----------:|')
+                        mark_down.write('\n')
+                        for param in getattr(item, container_name).values():
+                            mark_down.write('|`' + param.name + '`|')
+                            if isinstance(item, (Struct, Function)):
+                                mark_down.write('{}|{}|'.format(extract_type(param), str(param.is_mandatory)))
+                            mark_down.write(re.sub(r'\s{2,}', ' ', ''.join(param.description)).strip() + '|\n')
+                    mark_down.write('\n\n')
+                if i < len(interface) - 2:
+                    mark_down.write('\n<div style="page-break-after: always;"></div>\n\n')
+    except ParseError as error1:
+        print(error1)
+        sys.exit(1)
+
+
+def filter_pattern(interface, pattern):
+    """
+    Filtering Interface to extract only items matched with regex pattern
+    :param interface: Interface model
+    :param pattern: regex
+    :return:
+    """
+    match = {i: {} for i in vars(interface).keys()}
+    match['params'] = interface.params
+    if pattern:
+        for key, value in vars(interface).items():
+            if key == 'params':
+                continue
+            for name, item in value.items():
+                if re.match(pattern, item.name):
+                    if key in match:
+                        match[key].update({name: item})
+    else:
+        return vars(interface)
+    return match
+
+
+def main():
+    """
+    Main functions calls
+    """
+    args = get_parser()
+    if args.output_directory.exists() and args.skip:
+        print('Skipping {}'.format(args.output_directory))
+        return
+    if args.output_directory.exists() and not args.skip and not args.overwrite:
+        print('Exist {}, and skip or overwrite argument not provided'.format(args.output_directory))
+        return
+    if args.output_directory.exists() and args.overwrite:
+        print('Overwriting {}'.format(args.output_directory))
+    elif not args.output_directory.exists():
+        print('Creating new {}'.format(args.output_directory))
+
+    interface = Parser().parse(args.source_xml)
```

Comment:
Can the XSD file be passed to the parse function?

Reply:
@vladmu

Reply:
yes: `interface = Parser().parse(args.source_xml, **args.source_xsd**)`